src/test/java/core/di/factory/ComponentScanTest.java

```diff
@@ -0,0 +1,44 @@
+package core.di.factory;
+
+import com.google.common.collect.Sets;
+import core.annotation.Repository;
+import core.annotation.Service;
+import core.annotation.web.Controller;
+import core.di.factory.example.MyQnaService;
+import core.di.factory.example.QnaController;
+import next.reflection.ReflectionTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.reflections.Reflections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ComponentScanTest {
+    private static final Logger log = LoggerFactory.getLogger(ComponentScanTest.class);
+
+    private Reflections reflections;
+
+    @Test
+    public void printAllAnnotatedClasses() {
+        reflections = new Reflections("core.di.factory.example");
+        Set<Class<?>> annotatedClasses = getTypesAnnotatedWith(Controller.class, Service.class, Repository.class);
+
+        annotatedClasses.stream()
+            .forEach(clazz -> {
+                log.debug("className: {}", clazz.getName());
+            });
```

Comment:
Iterable 인터페이스에서는 jdk 1.8 부터 추가된 default 함수인 forEach 를 제공하고 있습니다.
```suggestion
        annotatedClasses
            .forEach(clazz -> {
                log.debug("className: {}", clazz.getName());
            });
```